x-pack/plugins/infra/public/containers/logs/log_entries/index.ts

```diff
@@ -124,48 +144,99 @@ const useFetchEntriesEffect = (
   dispatch: Dispatch,
   props: LogEntriesProps
 ) => {
-  const { getLogEntriesAround, getLogEntriesBefore, getLogEntriesAfter } = useGraphQLQueries();
-
-  const [prevParams, cachePrevParams] = useState(props);
+  const [prevParams, cachePrevParams] = useState<LogEntriesProps | undefined>();
   const [startedStreaming, setStartedStreaming] = useState(false);
 
-  const runFetchNewEntriesRequest = async (override = {}) => {
+  const runFetchNewEntriesRequest = async (overrides: Partial<LogEntriesProps> = {}) => {
+    if (!props.startTimestamp || !props.endTimestamp) {
+      return;
+    }
+
     dispatch({ type: Action.FetchingNewEntries });
+
     try {
-      const payload = await getLogEntriesAround({
-        ...omit(props, 'jumpToTargetPosition'),
-        ...override,
-      });
+      const fetchArgs: LogEntriesRequest = {
+        sourceId: overrides.sourceId || props.sourceId,
+        startTimestamp: overrides.startTimestamp || props.startTimestamp,
+        endTimestamp: overrides.endTimestamp || props.endTimestamp,
+        query: overrides.filterQuery || props.filterQuery || undefined, // FIXME
+      };
+
+      if (props.timeKey) {
+        (fetchArgs as LogEntriesCenteredRequest).center = props.timeKey;
+      } else {
+        (fetchArgs as LogEntriesBeforeRequest).before = 'last';
+      }
+
+      const { data: payload } = await fetchLogEntries(fetchArgs);
       dispatch({ type: Action.ReceiveNewEntries, payload });
+
+      // Move position to the bottom if it's the first load.
+      // Do it in the next tick to allow the `dispatch` to fire
+      if (!props.timeKey && payload.bottomCursor) {
+        setTimeout(() => {
```

Comment:
This rings alarm bells for me, but I can't immediately see a better way. :thinking: Just thinking out loud: Could this be moved into a separate effect to avoid `setTimeout`?

Reply:
> ...but I can't immediately see a better way.

What about... [`setImmediate()`](https://www.youtube.com/watch?v=6zXDo4dL7SU)?

I'm not a big fan either of the `setTimeout()`, but I couldn't really come with a better way to do it.

Regarding using `useEffect`, we could do an effect that triggers when there's a change in `state.bottomCursor`.

To be honest after working at length with the `useLogEntriesState` hook I don't like the pattern of using `useEffect` to change state. One of the things where I struggled the most when reading the codebase was to understand how the hook works.

Most of the behaviour happens in the effects as a reaction to changes in external state. The fetching happens implicitly because an external agent changes the state in `useLogPositionState`. That change triggered the effects, which are then responsible of checking inside them if they should fire or not (i.e. if it's streaming, if the date range has changed, if the scroll is in a certain position, etc.)

The `setTimeout()` here is necessary because of the asynchronous nature of `dispatch()`, but what I like about it is that we call its content explicitly. A fresh reader will know exactly _when_ we move the position to the bottom: right after the fetch happens. If we wrap it in a `useEffect`, he or she will need to build more of the world in their head to make sense of it.

----

I don't think there's a good way to make this work without making a relatively big change. `useLogEntriesState` and `useLogPositionState` are tightly coupled and I think they could be better merged into `useLogStreamState` (unless they are used individually somewhere else).

But that's a different PR :)

Reply:
Makes sense. Do we want to track the follow-up work somewhere?

Reply:
Tracking it in here https://github.com/elastic/kibana/issues/59875